codex-rs/exec-server/src/posix/mcp_escalation_policy.rs

```diff
@@ -12,14 +14,10 @@ use crate::posix::escalate_protocol::EscalateAction;
 use crate::posix::escalation_policy::EscalationPolicy;
 use crate::posix::stopwatch::Stopwatch;
 
-/// This is the policy which decides how to handle an exec() call.
-///
-/// `file` is the absolute, canonical path to the executable to run, i.e. the first arg to exec.
-/// `argv` is the argv, including the program name (`argv[0]`).
-/// `workdir` is the absolute, canonical path to the working directory in which to execute the
-/// command.
-pub(crate) type ExecPolicy = fn(file: &Path, argv: &[String], workdir: &Path) -> ExecPolicyOutcome;
+/// In-memory execpolicy rules that drive how to handle an exec() call.
+pub(crate) type ExecPolicy = Arc<Policy>;
```

Comment:
I don't think we need or want this type alias anymore.

Also, do we need to `Arc` it?